.freeCodeCamp/tooling/parser.js

```diff
@@ -229,6 +229,8 @@ function parseMarker(marker, lesson) {
 export function* seedToIterator(seed) {
   const sections = seed.match(new RegExp(`#### --(((?!#### --).)*\n?)`, 'sg'));
   for (const section of sections) {
+    if (isForceFlag(section)) { continue; }
```

Comment:
When `seedToIterator` is called, it's already after determining if lesson should be seeded.

https://github.com/freeCodeCamp/freeCodeCampOS/blob/520544e8ab961c18566611497c2e12521987b7cd/.freeCodeCamp/tooling/lesson.js#L62-L69

Flag is still going to be in the `seed` though. Without handling it, `seedToIterator` will throw _Seed is malformed_.

Reply:
I probably need to document this better: https://opensource.freecodecamp.org/freeCodeCampOS/project-syntax.html#--force--

**A**

- `seedEveryLesson == true`
- On next lesson, lesson **is** seeded

**B**

- `seedEveryLesson == true && isForce == true`
- On next lesson, lesson **NOT** seeded

**C**

- `seedEveryLesson == false`
- On next lesson, lesson **NOT** seeded

**D**

- `seedEveryLesson == false && isForce == true`
- On next lesson, lesson **IS** seeded

---

With this (PR) code, the force flag acts as a _set false_ flag - when present, never seed.

Hope this clarifies

Reply:
A little bit, I thought it is supposed to _force_ seed, regardless of what is `seedEveryLesson`. I did not expect it to behave as _A - D_ scenarios are showing. Looking closer at the above code in `lesson.js` they are indeed consistent with it.

---

However unless I'm again looking at the wrong thing, this didn't change. 

For example, this seed:
````
### --seed--

#### --force--

#### --"../file_parent.js"--

```js
const off = false;
```

#### --"file_in_project.js"--

```js
const n = true;
```
````
will have three `sections`:
```
[
  '#### --force--\n\n',
  '#### --"../file_parent.js"--\n\n```js\nconst off = false;\n```\n\n',
  '#### --"file_in_project.js"--\n\n```js\nconst n = true;\n```\n'
]
```

Section containing `--force--` is discarded, yes, but only from trying to process it as a command/seed. `--force--` is still taken under consideration when checking `seedEveryLesson` and `isFlag` interaction.

Reply:
`--force--` affects the `seed` level, not on a per, individual seed basis

Reply:
- `seedLesson` happens to only be called in one location, and this location handles `isForce` logic.
- Therefore, `seedToIterator` does not need to handle `isForce` logic.

That being said, doing so sounds like brittle code - easy to forget about `isForce` and call `seedLesson` elsewhere.

For now, I am probably ok with that being the case.

**Action**

Remove this commented line of code calling `continue`.


```suggestion
```

Reply:
Sorry, but that's not right. At least not as a lone change. Removing it will result in _`Error: Seed is malformed`_ errors, whenever lesson with seed containing `--force--` is attempted to be seeded.

(After `seedEveryLesson` and `isForce` logic are checked) `seedToIterator` receives `seed` in which `--force--` is still present. In `seedToIterator`, `--force--` will:

1. Not match file regex.
2. Not match command regex.

If both 1. and 2. not match, then _`Seed is malformed`_ is thrown.

Actually, outside of the PR, this already can happen when manually resetting step, which has `--force--` in seed. Seeding lesson internally doesn't use `seedToIterator`, but resetting step does.

Reply:
Well, then the parser needs to be amended to allow for the force flag.

Reply:
Sorry gikf! I now realise what you did, and how this works 🤦‍♂️ 

You are correct. This makes sense now.

Reply:
No worries :+1: